mongo/client.go

```diff
@@ -348,6 +351,11 @@ func (c *Client) endSessions(ctx context.Context) {
 }
 
 func (c *Client) configure(opts *options.ClientOptions) error {
+	var defaultOptions = 0
+	if opts.MaxPoolSize == nil {
+		defaultOptions++
+		opts.SetMaxPoolSize(defaultMaxPoolSize)
+	}
```

Comment:
Optional: Consider moving this `MaxPoolSize` default handling block down to where the existing `if opts.MaxPoolSize != nil` block is to keep all the `MaxPoolSize` option logic in the same place.

Reply:
Default options (`maxPoolSize` currently) need to be set first so they can all be checked in `Validate()`, and `Validate()` needs to be called before actual value modifications underneath. Similarly, I used `SetMaxPoolSize()` to indicate it an external setting somehow.

Reply:
Setting here seems fine to me in order to call `Validate` after and error quickly.